server.go

```diff
@@ -517,8 +696,23 @@ func (p *sshFxpSetstatPacket) respond(svr *Server) responsePacket {
 		} else if gid, _, err = unmarshalUint32Safe(b); err != nil {
 		} else {
 			err = os.Chown(p.Path, int(uid), int(gid))
-		}
-	}
+			fileUID = &uid
+			fileGID = &gid
+		}
+	}
+	svr.reqCallback(RequestPacket{
+		Type: Setstat,
+		Path: p.Path,
+		Attributes: &Attributes{
+			Size:             fileSize,
+			UID:              fileUID,
+			GID:              fileGID,
+			Permissions:      (*fs.FileMode)(permissions),
```

Comment:
It is well known that `fs.FileMode` is not binary compatible with [`filexfer.FileMode`](https://pkg.go.dev/github.com/pkg/sftp@v1.13.5/internal/encoding/ssh/filexfer#FileMode)